### PR TITLE
feat: add llama2 colab notebook

### PR DESCRIPTION
**Pull request by xianml**

### CODE REVIEW

**Comment by xianml** [comment]
preview in https://colab.research.google.com/drive/158zwSM__zs0caehysLinxLkjY7_naqcK?usp=sharing

**Comment by aarnphm** [comment]
@parano any comments on this when you have cycle?

**Review by aarnphm — COMMENTED** [review]
one nit thing: Lets remove some of the outputs from cell 1.

For cell 3, lets separate out a download block and a serve block

For cell 9 lets use the deployment SDK?

```python
bentoml.deployment.create
```

For the second notebook, with the gptq example lets use `TheBloke` models for gptq, otherwise we can do dynamic quantization

**Review by aarnphm — COMMENTED** [review]
one nit thing: Lets remove some of the outputs from cell 1.

For cell 3, lets separate out a download block and a serve block

For cell 9 lets use the deployment SDK?

```python
bentoml.deployment.create
```

For the second notebook, with the gptq example lets use `TheBloke` models for gptq, otherwise we can do dynamic quantization

**Comment by xianml** [comment]
@aarnphm @parano any new comments here?

**Review by aarnphm — APPROVED** [review]
finally the gptq we should also remove the outputs. LGTM from me, cc @parano

**Comment by xianml** [comment]
> finally the gptq we should also remove the outputs. LGTM from me, cc @parano
fix. tks~~

**Review by aarnphm — APPROVED** [review]

**Review by aarnphm — COMMENTED** [review]
also what is this fLlama? should we just call it function-calling-llama-demo?

**Comment by xianml** [comment]
> also what is this fLlama? should we just call it function-calling-llama-demo?

ok

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+| bentoml-llama2-demo      	| <a href="https://colab.research.google.com/drive/158zwSM__zs0caehysLinxLkjY7_naqcK?usp=sharing" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> 	|
+| openllm-llama2-demo      	| <a href="https://colab.research.google.com/drive/1NLFaFNxNxVgLPVgRevGv3NeyzJadNw96?usp=sharing" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> 	|
+| openllm-llama2-gptq-demo 	| <a href="https://colab.research.google.com/drive/1_8CWyOEpMH0eQzY9l1wc1OevTJf0VVXb?usp=sharing" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> 	|
+| bentoml-fLlama-demo      	| <a href="https://colab.research.google.com/drive/1HOaFA1ogMDPalGw_e55VcfDhiiblzrFO?usp=sharing" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> 	|
```
Comment (aarnphm) [inline examples/README.md:8]:
```suggestion
| function-calling-llama-demo      	| <a href="https://colab.research.google.com/drive/1HOaFA1ogMDPalGw_e55VcfDhiiblzrFO?usp=sharing" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> 	|
```

File: examples/function-calling-llama-demo/prompt.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            "enum": ["ON", "OFF", "UP", "DOWN", "SET_TEMPERATURE"],
+            "type": "string",
+        },
+        {
+            "name": "temperature",
+            "type": "number",
+            "description": "Only used together with the type argument is SET_TEMPERATURE",
+        }
+    ]
+}]
```
Comment (aarnphm) [inline examples/function-calling-llama-demo/prompt.py:29]: Let's use a different example here 😄

**Review by aarnphm — COMMENTED** [review]
@parano still need your input on this.

I think for the function-calling option we can use a different example fwiw.

File: examples/function-calling-llama-demo/prompt.py (hunk 2 above)
Comment (xianml) [reply examples/function-calling-llama-demo/prompt.py:29]: Okie. How about the example in hf? 
https://huggingface.co/Trelis/Llama-2-7b-chat-hf-function-calling-v2

```
"functionMetadata": {
        "function": "search_bing",
        "description": "Search the web for content on Bing. This allows users to search online/the internet/the web for content.",
        "arguments": [
            {
                "name": "query",
                "type": "string",
                "description": "The search query string"
            }
        ]
    }
'''
```

**Comment by xianml** [comment]
all colab examples are available in our google drive https://drive.google.com/drive/folders/1xxpJJaKISS9lxOI4SjL8CeqF2Zt5Vf_Q

File: examples/function-calling-llama-demo/prompt.py (hunk 2 above)
Comment (aarnphm) [reply examples/function-calling-llama-demo/prompt.py:29]: Yep that works

**Comment by aarnphm** [comment]
> all colab examples are available in our google drive [drive.google.com/drive/folders/1xxpJJaKISS9lxOI4SjL8CeqF2Zt5Vf_Q](https://drive.google.com/drive/folders/1xxpJJaKISS9lxOI4SjL8CeqF2Zt5Vf_Q)

Can we also make this publicly available to everyone? I believe currently this is only available internallly.

**Comment by aarnphm** [comment]
What is the status of this PR @xianml?

**Comment by xianml** [comment]
> What is the status of this PR @xianml?

Sorry. will split this PR to multi part for a quicker feedback loop. will deprecate this one

**Pull request closed without merging**